gfx_scene.h

```diff
@@ -244,22 +245,22 @@ GfxMetadata const &gfxSceneGetImageMetadata(GfxScene scene, uint64_t image_handl
 //! Image helpers.
 //!
 
-inline DXGI_FORMAT gfxImageGetFormat(GfxImage const &image)
-{
-    if(image.bytes_per_channel != 1 && image.bytes_per_channel != 2 && image.bytes_per_channel != 4) return DXGI_FORMAT_UNKNOWN;
-    uint32_t const bits = (image.bytes_per_channel << 3);
-    switch(image.channel_count)
-    {
-    case 1:
-        return (bits == 8 ? DXGI_FORMAT_R8_UNORM       : bits == 16 ? DXGI_FORMAT_R16_UNORM          : DXGI_FORMAT_R32_FLOAT         );
-    case 2:
-        return (bits == 8 ? DXGI_FORMAT_R8G8_UNORM     : bits == 16 ? DXGI_FORMAT_R16G16_UNORM       : DXGI_FORMAT_R32G32_FLOAT      );
-    case 4:
-        return (bits == 8 ? DXGI_FORMAT_R8G8B8A8_UNORM : bits == 16 ? DXGI_FORMAT_R16G16B16A16_UNORM : DXGI_FORMAT_R32G32B32A32_FLOAT);
-    default:
-        break;
+inline bool gfxImageIsFormatCompressed(GfxImage const& image)
+{
+    if (image.format == DXGI_FORMAT_BC1_TYPELESS || image.format == DXGI_FORMAT_BC1_UNORM ||
+        image.format == DXGI_FORMAT_BC1_UNORM_SRGB || image.format == DXGI_FORMAT_BC2_TYPELESS ||
+        image.format == DXGI_FORMAT_BC2_UNORM || image.format == DXGI_FORMAT_BC2_UNORM_SRGB ||
+        image.format == DXGI_FORMAT_BC3_TYPELESS || image.format == DXGI_FORMAT_BC3_UNORM ||
+        image.format == DXGI_FORMAT_BC3_UNORM_SRGB || image.format == DXGI_FORMAT_BC4_TYPELESS ||
+        image.format == DXGI_FORMAT_BC4_UNORM || image.format == DXGI_FORMAT_BC4_SNORM ||
+        image.format == DXGI_FORMAT_BC5_TYPELESS || image.format == DXGI_FORMAT_BC5_UNORM ||
+        image.format == DXGI_FORMAT_BC5_SNORM || image.format == DXGI_FORMAT_BC6H_TYPELESS ||
+        image.format == DXGI_FORMAT_BC6H_UF16 || image.format == DXGI_FORMAT_BC6H_SF16 ||
+        image.format == DXGI_FORMAT_BC7_TYPELESS || image.format == DXGI_FORMAT_BC7_UNORM ||
+        image.format == DXGI_FORMAT_BC7_UNORM_SRGB)             {
+        return true;
```

Comment:
Added an additional public function to check if a format is compressed or not (which is useful in various places)

Reply:
Sounds good, really wish MS would supply some ready-made helpers/macros for these...